includes/Media.php

```diff
@@ -204,6 +207,112 @@ public function init() {
+		add_filter( 'ajax_query_attachments_args', [ $this, 'filter_ajax_query_attachments_args' ] );
+		// Hide video posters from Media list view.
+		add_filter( 'pre_get_posts', [ $this, 'filter_poster_attachments' ] );
+		// Hide video posters from web-stories/v1/media REST API requests.
+		add_filter( 'rest_attachment_query', [ $this, 'filter_rest_poster_attachments' ], 10, 2 );
+	}
+
+	/**
+	 * Returns the tax query needed to exclude generated video poster images.
+	 *
+	 * @param array $args Existing WP_Query args.
+	 *
+	 * @return array  Tax query arg.
+	 */
+	private function get_poster_tax_query( array $args ) {
+		$tax_query = [
+			[
+				'taxonomy' => self::STORY_MEDIA_TAXONOMY,
+				'field'    => 'slug',
+				'terms'    => [ 'poster-generation' ],
+				'operator' => 'NOT IN',
+			],
+		];
+
+		/**
+		 *  Merge with existing tax query if needed,
+		 * in a nested way so WordPress will run them
+		 * with an 'AND' relation. Example:
+		 *
+		 * [
+		 *   'relation' => 'AND', // implicit.
+		 *   [ this query ],
+		 *   [ [ any ], [ existing ], [ tax queries] ]
+		 * ]
+		 */
+		if ( ! empty( $args['tax_query'] ) ) {
+			$tax_query[] = $args['tax_query'];
+		}
+
+		return $tax_query;
+	}
+
+	/**
+	 * Filters the attachment query args to hide generated video poster images.
+	 *
+	 * Reduces unnecessary noise in the Media grid view.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @param array $args Query args.
+	 *
+	 * @return array Filtered query args.
+	 */
+	public function filter_ajax_query_attachments_args( array $args ) {
+		$args['tax_query'] = $this->get_poster_tax_query( $args ); // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+
+		return $args;
+	}
+
+	/**
+	 * Filters the current query to hide generated video poster images.
+	 *
+	 * Reduces unnecessary noise in the Media list view.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param WP_Query $query WP_Query instance, passed by reference.
+	 *
+	 * @return void
+	 */
+	public function filter_poster_attachments( &$query ) {
+		$current_screen = function_exists( 'get_current_screen' ) ? get_current_screen() : null;
+
+		if ( ! $current_screen instanceof WP_Screen ) {
+			return;
+		}
+
+		if ( is_admin() && $query->is_main_query() && 'upload' === $current_screen->id ) {
+			$tax_query = $query->get( 'tax_query' );
+
+			$query->set( 'tax_query', $this->get_poster_tax_query( [ 'tax_query' => $tax_query ] ) ); // phpcs:ignore WordPress.DB.SlowDBQuery.slow_db_query_tax_query
+		}
+	}
+
+	/**
+	 * Filters the current query to hide generated video poster images.
+	 *
+	 * Reduces unnecessary noise in media REST API requests.
+	 *
+	 * @since 1.3.0
+	 *
+	 * @param array           $args Query args.
+	 * @param WP_REST_Request $request The current REST request.
+	 *
+	 * @return array Filtered query args.
+	 */
+	public function filter_rest_poster_attachments( array $args, WP_REST_Request $request ) {
+		if ( '/web-stories/v1/media' !== $request->get_route() ) {
+			return $args;
+		}
```

Comment:
My only worry about doing this like this, means if I build a little app that uses the media endpoint, I would see the poster images in that. In my testing yoast wasn't effected, but doesnt mean other plugins would not be.

Reply:
We could also hide it for the `wp/v2/media` endpoint to cover the majority of use cases (media library grid view, media library list view, REST API endpoints)

Reply:
In it worth noting, that in my testing, I couldn't find a place that used the media endpoint. Most of the media picker dialog still used admin ajax. 

So on second thoughts, let's leave it as it is and come to the issue if it comes up, if at all.